juniper_official/Linecard/check-pfe-discards.rule

```diff
@@ -0,0 +1,321 @@
+             * 
+             * Use FPC and PFE names as key for rule.
+             */  
+            keys [ fpc-no pfe-no ];
+            synopsis "Packet forwarding engine hardware discard statistics analyzer";
+            description "Collects packet forwarding engine hardware discard statistics  and notifies when discard count increases";
+            sensor pfe-sensor {
+                synopsis "iAgent netconf sensor definition";
+                description "Collects data from network device netconf rpc using get-pfe-traffic-statistics";
+                iAgent {
+                    file pfe-discard-statistics.yml;
+                    table PfeDiscardStatisticsTable;
+                    frequency 60s;
+                }
+            }
+            field drop-threshold {
+                constant {
+                    value "{{discard-count}}";
+                }
+                type integer;
+                description "Drop count increase threshold";
+            }
+            field fpc-no {
+                sensor pfe-sensor {
+                    path pfe-fpc-index;
+                }
+                type string;
+                description "FPC slot number";
+            }
+            field pfe-no {
+                sensor pfe-sensor {
+                    path pfe-index;
+                }
+                type string;
+                description "PFE number";
+            }
+            trigger data-error-discard {
+                synopsis "PFE data error discards KPI";
+                description "Sets health based on the change in PFE data error discard count";
+                frequency 60s;
+                term is-data-error-discard {
+                    when {
+                        increasing-at-least-by-value "$data-error-discard" {
+                            value "$drop-threshold";
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Increasing data-error-discard count $data-error-discard on FPC$fpc-no PFE$pfe-no";
+                        }
+                    }
+                }
+                term Term_2 {
+                    then {
+                        status {
+                            color green;
+                            message "Data-error-discard count $data-error-discard on FPC$fpc-no PFE$pfe-no is normal";
+                        }
+                    }
+                }
+            }
+            trigger pfe-bad-route-discard {
+                synopsis "PFE bad route discards KPI";
+                description "Sets health based on the change in PFE bad route discard count";
+                frequency 60s;
+                term is-bad-route-discard {
+                    when {
+                        increasing-at-least-by-value "$bad-route-discard";
```

Comment:
Should we be using a time-range here?

Reply:
Discards should be evaluated every time(sensor frequency) we receives data as these are critical hence time-range is not required.